tests/generate.py

```diff
@@ -527,5 +527,107 @@ def bad_algorithms_for_key(test_name, cert, unusable_algs):
     output.close()
 
 
+def client_auth():
+    def test(test_name,
```

Comment:
If the majority of `client_auth()` is the `test()` function, I'd prefer that `test()` moves out of the `client_auth()` scope, rename it to something like `generate_client_auth_test()` or something? It looks like it doesn't need extensive access to things that it's getting from its outer scope?